app/controllers/DownloadMetadataController.scala

```diff
@@ -38,19 +39,42 @@ class DownloadMetadataController @Inject() (
     }
   }
 
-  def downloadMetadataCsv(consignmentId: UUID): Action[AnyContent] = standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+  def downloadMetadataFile(consignmentId: UUID): Action[AnyContent] = standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
     for {
       metadata <- consignmentService.getConsignmentFileMetadata(consignmentId, request.token.bearerAccessToken, None, None)
       customMetadata <- customMetadataService.getCustomMetadata(consignmentId, request.token.bearerAccessToken)
       displayProperties <- displayPropertiesService.getDisplayProperties(consignmentId, request.token.bearerAccessToken, None, showInactive = true)
     } yield {
       val parseFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd[ ]['T']HH:mm:ss[.SSS][.SS][.S]")
       val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
-      val systemValues = List(fileUUID, fileName, clientSideOriginalFilepath, clientSideFileLastModifiedDate)
-      val nameMap = displayProperties.filter(dp => dp.active || systemValues.contains(dp.propertyName)).map(dp => (dp.propertyName, dp.displayName)).toMap
-      val filteredMetadata: List[CustomMetadata] = customMetadata.filter(cm => nameMap.keySet.contains(cm.name) && cm.allowExport).sortBy(_.exportOrdinal.getOrElse(Int.MaxValue))
+
+      val columnOrder = Seq(
+        fileUUID,
+        clientSideOriginalFilepath,
+        fileName,
+        clientSideFileLastModifiedDate,
+        end_date,
+        description,
+        former_reference,
+        closureType.name,
+        closureStartDate,
+        closurePeriod,
+        foiExemptionCode,
+        foiExemptionAsserted,
+        titleClosed,
+        titleAlternate,
+        descriptionClosed,
+        descriptionAlternate,
+        language,
+        filenameTranslated
+      )
+
+      val nameMap = displayProperties.filter(dp => columnOrder.contains(dp.propertyName)).map(dp => (dp.propertyName, dp.displayName)).toMap
+      val filteredMetadata: List[CustomMetadata] = columnOrder.collect(customMetadata.map(cm => cm.name -> cm).toMap).toList
       val header: List[String] = filteredMetadata.map(f => nameMap.getOrElse(f.name, f.name))
-      val fileMetadataRows: List[List[String]] = metadata.files.map { file =>
+      val dataTypes: List[DataType] = filteredMetadata.map(f => f.dataType)
+
+      val fileMetadataRows: List[List[String]] = metadata.files.sortBy(f => f.fileMetadata.find(_.name == clientSideOriginalFilepath).map(_.value.toUpperCase)).map { file =>
```

Comment:
Instead of creating `List[List[String]]`, create a `List[List[Any]]` then you don't need `dataTypes` to convert values in the app/controllers/util/ExcelUtils.scala. See my comment below.